### PR TITLE
Add page translation feature

### PR DESCRIPTION
**Pull request by sabertooth-cat**
Note: apertium-apy has also been modified.

https://codein.withgoogle.com/dashboard/task-instances/5480355175858176/?sp-page=1

### CODE REVIEW

File: assets/js/translator.js [hunk 1]
```diff
@@ -72,11 +74,12 @@ if(modeEnabled('translation')) {
             timer = setTimeout(function () {
                 if($('#instantTranslation').prop('checked')) {
                     translateText();
+                    
                 }
                 persistChoices('translator', true);
             }, timeout);
         });
-
+
```
Comment (sushain97) [inline assets/js/translator.js]: Remove trailing whitespace

File: assets/js/translator.js [hunk 2]
```diff
@@ -446,6 +482,49 @@ function translateText() {
...
+                    ajaxComplete();
+                    textTranslateRequest = undefined;
+                },
+                data: {
+                    'langpair': curSrcLang + '|' + curDstLang,
+                    'url': $('#url').val()
+                },
+                success: function (data) {
+                    if(data.responseStatus === 200) {
+                        var iframe = document.getElementById('translatedPage'),
```
Comment (sushain97) [inline assets/js/translator.js]: Use jQuery

File: assets/js/translator.js [hunk 3]
```diff
@@ -434,6 +460,16 @@ function translateText() {
                     if(data.responseStatus === 200) {
                         $('#translatedText').html(data.responseData.translatedText);
                         $('#translatedText').removeClass('notAvailable text-danger');
+                        var urlPattern = /(http|ftp|https):\/\/[\w-]+(\.[\w-]+)+([\w.,@?^=%&amp;:\/~+#-]*[\w@?^=%&amp;\/~+#-])?/
+                        if (urlPattern.test($('#originalText').val())) {
+                            $('#translatePageInstead').show();                            
+                            $("#translatePageInstead").attr("onclick","$('#translateText').hide(); $('#translatePage').show(); translateLink(\'".concat(data.responseData.translatedText.concat("')")));
```
Comment (sushain97) [inline assets/js/translator.js]: Single quotes

File: assets/css/style.css [hunk 4]
```diff
@@ -239,6 +239,29 @@ textarea#originalText, textarea#translatedText {
     }
 }
 
+iframe#translatedPage {
+    width:100%;
```
Comment (sushain97) [inline assets/css/style.css]: Consistent space between attributes and values

**Comment by unhammer** [comment]
I merged the apy part and approved the task; this does work, but I think there are some more things we could do before merging the html part into master, e.g.: 
1. detect url in text field (like google) instead of having a special button?
2. give the translated page more space

**Pull request closed without merging**